docs/src/main/asciidoc/cache.adoc

```diff
@@ -366,6 +378,112 @@ public class CachedService {
 <3> Calling this method WILL invalidate values cached by the `load` method because the key elements order is the same.
 <4> Calling this method WILL NOT invalidate values cached by the `load` method because the key elements order is different.
 
+=== Generating a cache key with `CacheKeyGenerator`
+
+You may want to include more than the arguments of a method into a cache key.
+This can be done by implementing the `io.quarkus.cache.CacheKeyGenerator` interface and declaring that implementation in the `keyGenerator` field of a `@CacheResult` or `@CacheInvalidate` annotation.
+
+If a CDI scope is declared on a key generator, then it will be injected as a CDI bean during the cache key computation.
+Otherwise, the key generator will be instantiated using its default constructor.
```

Comment:
I think this sentence is redundant given the warning below

Reply:
I think this sentence is more important than the warning. Should I remove the warning? The users will quickly understand what's wrong with the build time exception anyway.

Reply:
Agreed

Reply:
Warning removed.